### PR TITLE
BKNDLSS-33446 [Leaflet map] App crash after adding the invalid circle JSON in settings.

### PR DESCRIPTION
**Pull request by addKiwi**

### CODE REVIEW

File: components/bl-leaflet-map/src/helpers/map.js [hunk 1]
```diff
@@ -86,21 +86,35 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
   });
 }
 
+function isCircleDataValid(numbers, text, radius) {
+  const isNumbers = numbers.every(value => !isNaN(value));
+  const isText = text.every(value => typeof value === "string");
+  const isRadius = radius > 0;
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:92]: isRadiusValid or hasRadius

File: components/bl-leaflet-map/src/helpers/map.js [hunk 2]
```diff
@@ -86,21 +86,35 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+            onCircleClick({
+              coordinates: [lat, lng],
+              radius     : radius,
+              description: description,
+            });
+          })
+          .addTo(map)
+          .bindPopup(description);
+      } else {
+        console.warn('Circle data is not valid!');
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:116]: That looks like an error. Also, need extend error message with invalidation reason

File: components/bl-leaflet-map/src/helpers/map.js [hunk 3]
```diff
@@ -86,21 +86,45 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
 }
 
+function validateCircle(circle) {
+  const { point: { lat, lng }, radius, description } = circle;
+  const isNumbers = [lat, lng, radius].every(value => !isNaN(value));
+  const isText = description === undefined || typeof description === 'string';
+  const hasRadius = radius > 0;
+
+  if (!isNumbers) {
+    console.error('Circle coordinates and/or radius are not a number!\n', circle);
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:96]: Non-informative, we should write what we are waiting for and what we receive

File: components/bl-leaflet-map/src/helpers/map.js [hunk 4]
```diff
@@ -86,21 +86,45 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+  }
+
+  if (!isText) {
+    console.error('Circle description should be text!', circle);
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:100]: same

File: components/bl-leaflet-map/src/helpers/map.js [hunk 5]
```diff
@@ -86,21 +86,45 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+  }
+
+  if (isNumbers && !hasRadius) {
+    console.error('Circle radius should be greater than 0!', circle);
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:104]: same

File: components/bl-leaflet-map/src/helpers/map.js [hunk 6]
```diff
@@ -86,21 +86,45 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
   });
 }
 
+function validateCircle(circle) {
+  const { point: { lat, lng }, radius, description } = circle;
+  const isNumbers = [lat, lng, radius].every(value => !isNaN(value));
+  const isText = description === undefined || typeof description === 'string';
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:92]: Does not reflect what is happening

File: components/bl-leaflet-map/src/helpers/map.js [hunk 7]
```diff
@@ -86,21 +86,45 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
   });
 }
 
+function validateCircle(circle) {
+  const { point: { lat, lng }, radius, description } = circle;
+  const isNumbers = [lat, lng, radius].every(value => !isNaN(value));
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:91]: Does not reflect what is happening

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-leaflet-map/src/helpers/map.js [hunk 8]
```diff
@@ -86,21 +86,49 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+      console.error(`Circle error!\n Expected number but received "${ value }" in\n`, circle);
+    }
+
+    return result;
+  });
+  const isDescriptionIsText = description === undefined || typeof description === 'string';
+  const hasRadius = radius > 0;
+
+  if (!isDescriptionIsText) {
+    console.error(`Circle Erorr!\n Expected description type text but received "${ description }" in\n`, circle);
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:104]: Where do we use type "text" in codeless?

File: components/bl-leaflet-map/src/helpers/map.js [hunk 9]
```diff
@@ -86,21 +86,49 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+  }
+
+  if (isAllPropertiesAreNumbers && !hasRadius) {
+    console.error(`Circle error!\n Circle radius should be greater than 0, but received ${ radius } in\n`, circle);
+  }
+
+  return isAllPropertiesAreNumbers && isDescriptionIsText && hasRadius;
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:111]: we doesn't need all this variables just return false on the first validation error

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-leaflet-map/src/helpers/map.js [hunk 10]
```diff
@@ -86,21 +86,61 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+
+    return result;
+  });
+
+  if (!isAllPropertiesAreNumbers) {
+    return false;
+  }
+
+  if (description !== undefined && typeof description !== 'string') {
+    console.error(`Circle Erorr!\n Expected description type string but received "${ description }" in\n`, circle);
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:112]: 101 -> Circle error
112 -> Circle Erorr
118 -> Circle error

Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:112]: Expected - <type>, but received - <value>. Maybe we need change it to "receiver <type>"?

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-leaflet-map/src/helpers/map.js [hunk 11]
```diff
@@ -86,21 +86,61 @@ export function initMap(component, eventHandlers, map, currentLayer, uid) {
...
+    if (!result) {
+      console.error(`Circle error!\n Expected number but received "${ value }" in\n`, circle);
+    }
+
+    return result;
+  });
+
+  if (!isAllPropertiesAreNumbers) {
+    return false;
+  }
```
Comment (v-excelsior) [inline components/bl-leaflet-map/src/helpers/map.js:109]: forEach?

Comment (addKiwi) [reply components/bl-leaflet-map/src/helpers/map.js:109]: The foreEach method returns only `undefined`.

Comment (v-excelsior) [reply components/bl-leaflet-map/src/helpers/map.js:109]: We doesn't need return. forEach - isn't what we need in any case. Use `for of` to iterate and break validating at this step.

**Pull request merged**